### PR TITLE
chore: add new chain B2 Mainnet

### PR DESCRIPTION
**Pull request by oxf71**
# Add New Chain <chainId>

Thanks for your pull request to add a new support in Sourcify.

If you haven't done so, please follow the instructions on [how to request chain support](https://docs.sourcify.dev/docs/chain-support/) in docs.

Please check the following items before submitting your pull request for a speedy review.

## Checklist

- [x] The branch is named as `add-chain-<chainId>`.
- [x] I haven't modified the `chains.json` file directly.
- [x] In `sourcify-chains.json` file
  - [x] I've set `supported: true`.
  - [ ] I haven't added an `rpc` field but the one in [chains.json](../../src/chains.json) is used (if not, please explain why).
- [x] I've added a test in [chain-tests.js](../../test/chains/chains-test.js) file.
- [x] `test-new-chain` test in Circle CI is passing.

### CODE REVIEW

**Comment by oxf71** [comment]
@manuelwedler

File: services/server/src/sourcify-chains-default.json [hunk 1]
```diff
@@ -1892,5 +1892,21 @@
...
+    "etherscanApi": {
+      "apiURL": "https://explorer.bsquared.network/api"
+    },
+    "fetchContractCreationTxUsing": {
+      "blockscoutApi": {
+        "url": "https://mainnet-backend.bsquared.network"
+      },
+      "routescanApi": {
+        "type": "mainnet"
+      },
```
Comment (manuelwedler) [inline services/server/src/sourcify-chains-default.json:1908]: I cannot find your network on https://routescan.io/. Is Routescan really supporting it?

File: services/server/src/sourcify-chains-default.json [hunk 2]
```diff
@@ -1892,5 +1892,21 @@
     "sourcifyName": "TixChain Testnet",
     "supported": true,
     "rpc": ["https://subnets.avax.network/tixchain/testnet/rpc"]
+  },
+  "223": {
+    "sourcifyName": "B2 Mainnet",
+    "supported": true,
+    "etherscanApi": {
+      "apiURL": "https://explorer.bsquared.network/api"
```
Comment (manuelwedler) [inline services/server/src/sourcify-chains-default.json:1900]: Is this also supporting the `/api?module=contract&action=getcontractcreation`? I cannot find it in the documentation. Would we need an API key for using it?

**Comment by manuelwedler** [comment]
It seems the etherscan API is not fully implemented. I tried the `getsourcecode` method with https://explorer.bsquared.network/api?module=contract&action=getsourcecode&contractaddresses=0x2b7aa8bdc40b6d3d19d0de7480c4db8d5b6495e2 but I get 500 error back

**Review by manuelwedler — APPROVED** [review]

**Comment by oxf71** [comment]
> It seems the etherscan API is not fully implemented. I tried the `getsourcecode` method with https://explorer.bsquared.network/api?module=contract&action=getsourcecode&contractaddresses=0x2b7aa8bdc40b6d3d19d0de7480c4db8d5b6495e2 but I get 500 error back

Thank you very much, I will give feedback to our explorer team

**Pull request merged**